edx_arch_experiments/codejail_service/tests/test_views.py

```diff
@@ -0,0 +1,149 @@
+"""
+Test codejail service views.
+"""
+
+import json
+import textwrap
+from os import path
+
+import ddt
+from django.contrib.auth import get_user_model
+from django.test import TestCase, override_settings
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+from edx_arch_experiments.codejail_service import views
+
+
+@override_settings(
+    ROOT_URLCONF='edx_arch_experiments.codejail_service.urls',
+    MIDDLEWARE=[
+        'django.contrib.sessions.middleware.SessionMiddleware',
+    ],
+)
+@ddt.ddt
+class TestExecService(TestCase):
+    """Test the v0 code exec view."""
+
+    def setUp(self):
+        super().setUp()
+        user_model = get_user_model()
+        self.admin_user = user_model.objects.create_user('cms_worker', is_staff=True)
+        self.other_user = user_model.objects.create_user('student', is_staff=False)
+        self.standard_params = {'code': 'retval = 3 + 4', 'globals_dict': {}}
+
+    def _test_codejail_api(self, *, user=None, files=None, skip_auth=False, params=None, exp_status, exp_body):
+        """
+        Call the view and make assertions.
+
+        Arguments:
+            user: User to authenticate as when calling view (None for unauthenticated)
+            exp_status: Assert that the response HTTP status code is this value
+            exp_body: Assert that the response body JSON is this value
```

Comment:
missing a few arguments here

Reply:
Thanks, fixed. And one of the lines was a lie, too.